src/main.rs

```diff
@@ -3106,11 +3111,118 @@ impl RicCall {
                     return bad_argument(req_id, json, format!("cannot delete the net peering {}", net_peering_id).as_str())
                 }
  
-                Ok((jsonobj_to_strret(json, req_id), StatusCode::OK))
+            Ok((jsonobj_to_strret(json, req_id), StatusCode::OK))
+        },
+        RicCall::CreateVirtualGateway => {
+            if auth != AuthType::AkSk {
+                return eval_bad_auth(req_id, json, "CreateVirtualGateway require v4 signature")
+            }
+            let in_json = require_in_json!(bytes);
+            println!("{:#}", in_json.dump());
+
+            let connection_type = require_arg!(in_json, "ConnectionType");
+            let virtual_gateway = json::object!{
+                VirtualGatewayId: format!("vgw-{:08x}", req_id),
+                "ConnectionType": "ipsec.1",
+                "NetToVirtualGatewayLinks": [],
+                "State": "available",
+                "Tags": []
+            };
+
+            if connection_type != virtual_gateway["ConnectionType"] {
```

Comment:
```rust
if connection_type != "ipsec.1" {
```
Would be more logical here, as, it is what yo really want to check.

Reply:
yes it makes more sense